src/containers/SarcophagusCreate/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { Formik, FormikErrors } from "formik";
+import { useEffect, useState } from "react";
+import { useHistory } from "react-router";
+import {
+  createInitialValues,
+  createValidationSchema,
+} from "../../components/SarcophagusCreate/sarcophagusCreate.utils";
+import { useArchaeologistsStore } from "../../stores/Archaeologist";
+import { Archaeologist, ArchaeologistsStore } from "../../stores/Archaeologist/archaeologist.interfaces";
+import useApproval from "../../stores/BlockChain/useApproval";
+import { useSarcophagiStore } from "../../stores/Sarcophagi";
+import { SarcophagusStore } from "../../stores/Sarcophagi/sarcophagi.interfaces";
+import { useWeb3 } from "../../web3";
+import useFileEncryption from "../../components/SarcophagusCreate/hooks/useFileEncryption";
+import { BigNumber, utils } from "ethers";
+import { SarcophagusCreateValues } from "../../components/SarcophagusCreate/sarcophagusCreate.interfaces";
+import { connect } from "../../web3/providers";
+import CreateTitleAndDescription from "../../components/SarcophagusCreate/CreateTitleAndDescription";
+import Settings from "../../components/SarcophagusCreate/Settings";
+import UploadSarcophagusFile from "../../components/SarcophagusCreate/UploadSarcophagusFile";
+import SelectResurrectionTime from "../../components/SarcophagusCreate/SelectResurrectionTime";
+import ArchaeologistSelectForm from "../../components/SarcophagusCreate/SelectArchaeologistForm";
+import Button from "../../components/layout/Button";
+
+const CreateSarcophagus = () => {
+  const [buttonText, setButtonText] = useState("");
+  const [storageFee, setStorageFee] = useState<number | string | BigNumber | null>(null);
+  const [selectedArchaeologist, setSelectedArchaeologist] = useState<Archaeologist | null>(null);
+  const history = useHistory();
+
+  const {
+    file,
+    setFile,
+    setRecipientPublicKey,
+    setArchaeologistAddress,
+    doubleEncryptedFile,
+    assetDoubleHash,
+  } = useFileEncryption();
+
+  const sarcophagiStore: SarcophagusStore = useSarcophagiStore();
+  const archaeologistsStore: ArchaeologistsStore = useArchaeologistsStore();
+  const { account } = useWeb3();
+  const { approved, approveTransaction } = useApproval();
+
+  const handleArchaeologistSelect = (
+    selectedArchaeologist: Archaeologist,
+    storageFee: number | string | BigNumber
+  ) => {
+    setStorageFee(storageFee);
+    setSelectedArchaeologist(selectedArchaeologist);
+    setArchaeologistAddress(selectedArchaeologist?.currentPublicKey);
+  };
+
+  const successRefresh = () => {
+    sarcophagiStore.loadSarcophagi();
+    archaeologistsStore.loadArchaeologists();
+  };
+
+  const redirect = () => {
+    history.push("/tomb");
+  };
+
+  const createSarcophagus = async (values: SarcophagusCreateValues) => {
+    if (!selectedArchaeologist || !storageFee || !assetDoubleHash || !doubleEncryptedFile) {
+      console.error("🚀 ~ Create values not found");
+      return;
+    }
```

Comment:
It may help for debugging purposes to be more explicit in the logs about which one of these is null. Could log all of the vars out here if you don't want to change the conditional logic.

Also, what happens if this returns? Does the UX notify the user they have to take an action?